Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -69,52 +68,126 @@ extension FileSystem {
+        if idiomaticConfigurationDirectory != self.dotSwiftPMConfigurationDirectory {
+            // copy the configuration files from old location (eg /Users/<user>/Library/org.swift.swiftpm) to new one (eg /Users/<user>/Library/org.swift.swiftpm/configuration)
+            // but leave them there for backwards compatibility (eg older xcode)
+            let oldConfigDirectory = idiomaticConfigurationDirectory.parentDirectory
+            if self.exists(oldConfigDirectory, followSymlink: false) && self.isDirectory(oldConfigDirectory) {
+                let content = try self.getDirectoryContents(oldConfigDirectory).filter{ !$0.hasSuffix(".lock") }
+                for item in content {
+                    if self.isFile(oldConfigDirectory.appending(component: item)) &&
+                        !self.isSymlink(oldConfigDirectory.appending(component: item)) &&
+                        !self.exists(idiomaticConfigurationDirectory.appending(component: item)) {
+                        observabilityScope?.emit(warning: "Usage of \(oldConfigDirectory.appending(component: item)) has been deprecated. Please delete it and use the new \(idiomaticConfigurationDirectory.appending(component: item)) instead.")
+                        try self.copy(from: oldConfigDirectory.appending(component: item), to: idiomaticConfigurationDirectory.appending(component: item))
+                    }
+                }
+            }
+        // in the case where ~/.swiftpm/configuration is the idiomatic location (eg on Linux)
+        } else {
+            // copy the configuration files from old location (~/.swiftpm/config) to new one (~/.swiftpm/configuration)
+            // but leave them there for backwards compatibility (eg older toolchain)
+            let oldConfigDirectory = self.dotSwiftPM.appending(component: "config")
+            if self.exists(oldConfigDirectory, followSymlink: false) && self.isDirectory(oldConfigDirectory) {
+                let content = try self.getDirectoryContents(oldConfigDirectory).filter{ !$0.hasSuffix(".lock") }
+                for item in content {
+                    if self.isFile(oldConfigDirectory.appending(component: item)) &&
+                        !self.isSymlink(oldConfigDirectory.appending(component: item)) &&
+                        !self.exists(idiomaticConfigurationDirectory.appending(component: item)) {
+                        observabilityScope?.emit(warning: "Usage of \(oldConfigDirectory.appending(component: item)) has been deprecated. Please delete it and use the new \(idiomaticConfigurationDirectory.appending(component: item)) instead.")
+                        try self.copy(from: oldConfigDirectory.appending(component: item), to: idiomaticConfigurationDirectory.appending(component: item))
+                    }
+                }
+            }
+        }
+        // ~temporary 5.6 migration
+
+        // Create idiomatic if necessary
+        if !self.exists(idiomaticConfigurationDirectory) {
```

Comment:
We already have this? L93-95

Reply:
yea, but I'd like to leave it here so we can remove the whole block above and not forget to put that back in